pkg/util/podstore/podstore_test.go

```diff
@@ -0,0 +1,549 @@
+			Namespace: "pod1_ns",
+			UID:       "pod1",
+		},
+	}
+	tests := []struct {
+		name        string
+		oldObj      interface{}
+		newObj      interface{}
+		expectedPod *v1.Pod
+	}{
+		{
+			name:        "newObj is not Pod",
+			newObj:      node,
+			expectedPod: pod1,
+		},
+		{
+			name:        "valid case",
+			newObj:      newPod1,
+			expectedPod: newPod1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			podStore := &PodStore{
+				pods: cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+			}
+			err := podStore.pods.Add(pod1)
+			require.NoError(t, err)
+			podStore.onPodUpdate(tt.oldObj, tt.newObj)
+			require.Len(t, podStore.pods.List(), 1)
+			assert.Equal(t, tt.expectedPod, podStore.pods.List()[0].(*v1.Pod))
+		})
+	}
+}
+
+func Test_onPodCreate(t *testing.T) {
+	tests := []struct {
+		name        string
+		obj         interface{}
+		expectedMap map[types.UID]*podTimestamps
+	}{
+		{
+			name:        "object is not Pod",
+			obj:         node,
+			expectedMap: map[types.UID]*podTimestamps{},
+		},
+		{
+			name:        "valid case for Pending Pod",
+			obj:         pod1,
+			expectedMap: map[types.UID]*podTimestamps{"pod1": {CreationTimestamp: refTime}},
+		},
+		{
+			name:        "valid case for Running Pod",
+			obj:         pod3,
+			expectedMap: map[types.UID]*podTimestamps{"pod3": {CreationTimestamp: refTime.Add(-5 * time.Minute)}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			podStore := &PodStore{
+				timestampMap: map[types.UID]*podTimestamps{},
+				clock:        clock.NewFakeClock(refTime),
+				pods:         cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+			}
+			podStore.onPodCreate(tt.obj)
+			assert.Equal(t, tt.expectedMap, podStore.timestampMap)
+		})
+	}
+}
+
+func Test_onPodDelete(t *testing.T) {
+	k8sClient := fake.NewSimpleClientset()
+	podInformer := coreinformers.NewPodInformer(k8sClient, metav1.NamespaceAll, 0, cache.Indexers{})
+	fakeClock := clock.NewFakeClock(refTime)
+	podStore := NewPodStoreWithClock(podInformer, fakeClock)
+	err := podStore.pods.Add(pod1)
+	require.NoError(t, err)
+	podStore.timestampMap = map[types.UID]*podTimestamps{"pod1": {CreationTimestamp: refTime}}
+
+	t.Run("object is neither Pod or DeletedFinalStateUnknown", func(t *testing.T) {
+		podStore.onPodDelete(node)
+		assert.Equal(t, &podTimestamps{CreationTimestamp: refTime}, podStore.timestampMap["pod1"])
+	})
+
+	t.Run("Pod is in prevPod and podsToDelete", func(t *testing.T) {
+		expectedDeleteTime := refTime.Add(delayTime)
+		podStore.onPodDelete(pod1)
+		assert.Equal(t, &podTimestamps{CreationTimestamp: refTime, DeletionTimestamp: &refTime}, podStore.timestampMap["pod1"])
+		fakeClock.SetTime(expectedDeleteTime.Add(-10 * time.Millisecond))
+		assert.Eventuallyf(t, func() bool {
+			return podStore.podsToDelete.Len() == 0
+		}, 1*time.Second, 10*time.Millisecond, "Pod should not be added to PodsToDelete")
```

Comment:
using `assert.Eventuallyf` is quite meaningless here: `podStore.podsToDelete.Len()` is always going to be 0 on the first iteration. If you want, you can sleep for 100ms before checking.

Reply:
Removed. Thanks.